### PR TITLE
Add syslog, splunk, sumologic and new rewrelic integration linking support

### PR DESCRIPTION
**Pull request by Ndifreke**
Add Support for new integrations

### CODE REVIEW

File: src/model/Ressource.ts [hunk 1]
```diff
@@ -317,7 +317,7 @@ export default abstract class Ressource {
   }
 
   static wrap<T>(objs: T[]): T[] {
-    return objs.map((obj: APIObject) => getInstance(this, obj));
+    return objs.map((obj) => getInstance(this, obj));
```
Comment (shakaman) [inline src/model/Ressource.ts:320]: :thinking: why did you remove the type here?

File: src/model/Integration.ts [hunk 2]
```diff
@@ -15,9 +15,28 @@ const types = [
...
+  "facility",
+  "message_format",
+  "auth_mode",
+  "tls_verify",
+  "auth_token",
+
+  //splunk
+  "index",
+  "sourcetype",
+  "url",
```
Comment (shakaman) [inline src/model/Integration.ts:38]: this field is already defined below.

File: src/model/Ressource.ts (hunk 1 above)
Comment (Ndifreke) [reply src/model/Ressource.ts:320]: The type definition is wrong, for this to would have been valid, it should be. 
```
 static wrap<T>(objs: APIObject[]): T[] {
    return objs.map((obj: APIObject) => getInstance(this, obj));
  }
```
But since T[] is a generic of array and T has no constrainst to ``APIOject`` the return type will be deduced when` wrap<T>(ojs[]) `is expressed
